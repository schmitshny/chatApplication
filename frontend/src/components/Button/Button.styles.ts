import styled from 'styled-components';
import { ButtonProps } from './Button';

export const StyledButton = styled.button<ButtonProps>`
  padding: 10px 20px;
  border-radius: 4px;
  border: none;
  font-size: 16px;
  cursor: pointer;
  color: #eee;

  ${({ $variant }) =>
    $variant === 'primary' &&
    `
    background-color: #007bff;
    &:hover {
      background-color: #0056b3;
    }
  `}

  ${({ $variant }) =>
    $variant === 'secondary' &&
    `
    background-color: #FF69B4;
    &:hover {
      background-color: #FF41A1;
    }
  `}

   &:disabled {
    background-color: #ccc;
    cursor: not-allowed;
  }
`;
