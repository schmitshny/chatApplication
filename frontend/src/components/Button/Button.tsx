import React from 'react';
import { StyledButton } from './Button.styles';

export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  $variant?: 'primary' | 'secondary';
}

export const Button: React.FC<ButtonProps> = ({ $variant = 'primary', children, ...props }) => {
  return (
    <StyledButton $variant={$variant} {...props}>
      {children}
    </StyledButton>
  );
};
