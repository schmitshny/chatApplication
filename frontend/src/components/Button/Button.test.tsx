import { render, screen } from '@testing-library/react';
import { Button } from './Button';
import { describe, expect, test } from 'vitest';

describe('Button Component', () => {
  test('renders Button component', () => {
    const { container } = render(<Button>Click me</Button>);

    expect(screen.getByRole('button', { name: 'Click me' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test.each(['primary', 'secondary'] as const)('renders with %s variant', (variant) => {
    const { container } = render(<Button $variant={variant}>Click me</Button>);

    expect(container).toMatchSnapshot();
  });

  test('forwards additional props to the button element', () => {
    const { container } = render(<Button aria-label="Custom label">Click me</Button>);

    expect(screen.getByRole('button', { name: 'Custom label' })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
