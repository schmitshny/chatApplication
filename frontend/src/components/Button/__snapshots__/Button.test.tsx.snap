// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > forwards additional props to the button element 1`] = `
<div>
  <button
    aria-label="Custom label"
    class="sc-bczSft eOvKcj"
  >
    Click me
  </button>
</div>
`;

exports[`Button Component > renders Button component 1`] = `
<div>
  <button
    class="sc-bczSft eOvKcj"
  >
    Click me
  </button>
</div>
`;

exports[`Button Component > renders with primary variant 1`] = `
<div>
  <button
    class="sc-bczSft eOvKcj"
  >
    Click me
  </button>
</div>
`;

exports[`Button Component > renders with secondary variant 1`] = `
<div>
  <button
    class="sc-bczSft fTuyEI"
  >
    Click me
  </button>
</div>
`;
